logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -21,6 +21,10 @@ def initialize
       # We keep the structured cache to allow
       # the api to search the content of the differents nodes
       @store = Concurrent::Map.new
+
+      # This hash has only one dimension
+      # and allow fast retrieval of the metrics
+      @fast_lookup = Concurrent::Map.new
```

Comment:
I' sorry, for me is a copy not a pointer to the right map place, now you are saving two times the same data.